src/contrib/mlir/dialect/ops.td

```diff
@@ -182,6 +182,18 @@ def NGDotOp : NG_Binary_Op<"dot">
   let verifier = [{ return verifyOp(this); }];
 }
 
+// TODO(amprocte): Might be nice to rebase this on some sort of NG_Variadic_Op
+// class, but I'm not sure how to add concatenation_axis into the args if we
+// do that.
```

Comment:
Maybe doing something similar to `list<OpTrait> traits = []` below would work?

Reply:
You can try to do dag concat in the leaf class. 
`let arguments = !con(arguments, (ins I64Attr:$concatenation_axis))`
https://llvm.org/docs/TableGen/LangIntro.html

Reply:
Will give this a try

Reply:
Tablegen seems to interpret this as a recursive rebinding of a "arguments" or something... it complains about not being able to result `!con(!con(arguments, (ins I64Attr:$concatenation_axis)), (ins I64Attr:$concatenation_axis)))`. May be best to stick with this until someone who knows what they are doing with tablegen does a refactor. ;)

Reply:
There are subtleties about how the classes are expanded. I will address it later.